bindings/cpp/newapi/result_entry.cpp

```diff
@@ -167,4 +179,231 @@ data_pointer iterator_result_entry::data() const {
+
+	int sort() const
+	{
+		std::vector<struct iterator_container_item> items(m_num_items);
+		const size_t items_size = m_num_items * sizeof(iterator_container_item);
+
+		int err = dnet_read_ll(m_fd, reinterpret_cast<char *>(items.data()), items_size, m_offset);
+		if (err)
+			return err;
+
+		std::sort(items.begin(), items.end(), compare_iterator_container_items);
+
+		return dnet_write_ll(m_fd, reinterpret_cast<char *>(items.data()), items_size, m_offset);
+	}
+
+	const struct iterator_container_item &get_item() const
+	{
+		return m_buffer[m_buffer_index];
+	}
+
+	bool next()
+	{
+		if (++m_num_processed_items >= m_num_items)
+			return false;
+
+		if (++m_buffer_index >= m_buffer.size()) {
+			m_buffer_index = 0;
+			if (read_buffer() != 0)
+				return false;
+		}
+		return true;
+	}
+
+	int set_num_buffer_items(size_t num_buffer_items)
+	{
+		m_buffer.resize(num_buffer_items);
+		return read_buffer();
+	}
+
+private:
+	int read_buffer()
+	{
+		const auto remaining_items = std::min(m_buffer.size(), m_num_items - m_num_processed_items);
+		const auto buffer_size = remaining_items * sizeof(iterator_container_item);
+		const auto offset = m_offset + m_num_processed_items * sizeof(iterator_container_item);
+		return dnet_read_ll(m_fd, reinterpret_cast<char *>(m_buffer.data()), buffer_size, offset);
+	}
+
+private:
+	int m_fd;
+	const uint64_t m_offset;
+	const size_t m_num_items;
+
+	std::vector<iterator_container_item> m_buffer;
+	size_t m_buffer_index;
+	size_t m_num_processed_items;
+};
+
+struct ChunksComparator
+{
+	bool operator () (const iterator_result_chunk *lhs, const iterator_result_chunk *rhs) const
+	{
+		return compare_iterator_container_items(rhs->get_item(), lhs->get_item());
+	}
+};
+
+//* Append one result to container
+void iterator_result_container::append(const iterator_result_entry &result)
+{
+	if (m_sorted)
+		throw_error(-EROFS, "can't append to already sorted container");
+
+	dnet_iterator_response dnet_response;
+	deserialize(result.raw_data(), dnet_response);
+
+	iterator_container_item item;
+	item.key = dnet_response.key;
+	item.status = dnet_response.status;
+	item.record_flags = dnet_response.record_flags;
+	item.user_flags = dnet_response.user_flags;
+	item.json_timestamp = dnet_response.json_timestamp;
+	item.json_size = dnet_response.json_size;
+	item.json_capacity = dnet_response.json_capacity;
+	item.read_json_size = dnet_response.read_json_size;
+	item.data_timestamp = dnet_response.data_timestamp;
+	item.data_size = dnet_response.data_size;
+	item.read_data_size = dnet_response.read_data_size;
+	item.data_offset = dnet_response.data_offset;
+	item.blob_id = dnet_response.blob_id;
+
+	int err = dnet_write_ll(m_fd, reinterpret_cast<char *>(&item), sizeof(item), m_write_position);
+	if (err != 0)
+		throw_error(err, "dnet_write_ll failed");
+	m_write_position += sizeof(item);
+	m_count++;
+}
+
+//* Sort container by (key, timestamp) tuple
+void iterator_result_container::sort()
+{
+	if (m_sorted == true)
```

Comment:
Since it is in your PR, you are the real author, so please take a responsible and clean the code.